### ISSUE TITLE
Interoperate with "pyenv global" Python version by default

### ISSUE THREAD

**Issue by sfc-gh-afedorov** [pipxproject/pipx#654]
**Describe the bug**

I would like to use `pyenv global` with `pipx` to install `dbt-core==0.18.1b1` using `py38`, like so —

~~~
brew install pyenv pipx
eval "$(pyenv init -)
pyenv install 3.8.8
pyenv global 3.8.8
pipx install dbt-core==0.18.1b1
~~~

this does not work, perhaps because the `python` that pyenv uses is a shell script —

~~~
~ % which python
/Users/afedorov/.pyenv/shims/python
~ % cat `which python`
#!/usr/bin/env bash
set -e
[ -n "$PYENV_DEBUG" ] && set -x

program="${0##*/}"

export PYENV_ROOT="/Users/afedorov/.pyenv"
exec "/usr/local/Cellar/pyenv/1.2.24.1/libexec/pyenv" exec "$program" "$@"
~~~

instead we need to do —

~~~
pipx install dbt-core==0.18.1b1 --python 'python'
~~~

Which works, but can be confusing to folks on our team. This is not an enormous problem, but consideration and explanation of this functionality would be greatly appreciated.

**Comment by itsayellow** [comment]
Yes, we have been hoping to come up with a better solution for detecting the global default python automatically, but so far we don't have a great method. (See discussion in #638.)

One workaround that might be a little easier than using the `--python` switch for each install, would be to use the `PIPX_DEFAULT_PYTHON` environment variable.  If that's in your shell environment, pipx will use that python for all commands.

It is interesting that all you need to pass to the `--python` switch is the string `'python'`.  I would've thought you'd need a full path.

**Comment by trallnag** [comment]
I'm also using `PIPX_DEFAULT_PYTHON` without any issues. In the future another alternative will / might be providing a self contained pipx
